src/python/pants/engine/target.py

```diff
@@ -296,20 +296,10 @@ def field_types(self) -> Tuple[Type[Field], ...]:
 
     @final
     @memoized_classproperty
-    def PluginField(cls) -> Type:
-        """A sentinel class to allow plugin authors to add additional fields to this target type.
-
-        Plugin authors may add additional fields by simply registering UnionRules between the
-        `Target.PluginField` and the custom field, e.g. `UnionRule(PythonLibrary.PluginField,
-        TypeChecked)`. The `Target` will then treat `TypeChecked` as a first-class citizen and
-        plugins can use that Field like any other Field.
-        """
-
-        # NB: It's important that every subclass of `Target` have its own distinct `PluginField`
-        # type so that registering a PluginField for one target doesn't end up registering for all
-        # targets. This means that we cannot use a statically declared class, so we instead
-        # dynamically create a class. We use a class property and name the method in PascalCase so
-        # that this implementation doesn't leak to call sites.
+    def _anonymous_plugin_field_cls(cls) -> Type:
+        # NB: We make this a method, rather than using a static class, to ensure that every
+        # subclass has a distinct `.PluginField` class. This ensures that registering a plugin
```

Comment:
The comment has drifted. There is no longer method vs static class nor `.PluginField`. Perhaps just focus on the 1 union per Target subtype need:
```python
@final
@memoized_classproperty
def _plugin_field_union(cls) -> Type:
  # N.B.: We want each Target subtype to collect its own union of plugin fields to ensure plugin
  # fields don't leak across target types.
  return union(type(f"{cls.__name__}PluginField", (), {}))
```